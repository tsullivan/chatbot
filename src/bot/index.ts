export { Bot } from './bot';
export { RandomMessage } from './handle_chat/random_message';
export { Session, SessionProto } from './session';
