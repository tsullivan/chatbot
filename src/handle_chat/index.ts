export { handleChat } from './handle_chat';
export { RandomMessage } from './random_message';
