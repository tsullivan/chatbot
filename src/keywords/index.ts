import { KeywordResponder } from './keyword_responder';
import * as coinflip from './responders/coinflip';
import * as fact from './responders/fact';
import * as help from './responders/help';
import * as joke from './responders/joke';
import * as just_dont from './responders/just_dont';
import * as name from './responders/name';
import * as play from './responders/play';
import * as random from './responders/random';
import * as remember from './responders/remember';
import * as repeat from './responders/repeat';
import * as roll from './responders/roll';
import * as say from './responders/say';
import * as score from './responders/score';
import * as what from './responders/what';

export function keywordTester(input, chat) {
  const responders = getResponders();
  for (const keyword in responders) {
    if (responders.hasOwnProperty(keyword)) {
      try {
        const responder: KeywordResponder = new responders[keyword].KeywordResponder(input, chat);
        if (responder.inputMatches()) {
          return {
            isKeyword: true,
            responder,
          };
        }
      } catch (err) {
        throw new Error('Bad keyword responder constructor: ' + keyword);
      }
    }
  }
  return { isKeyword: false };
}

export function getResponders() {
  return {
    coinflip,
    help,
    joke,
    just_dont,
    name,
    fact,
    play,
    random,
    remember,
    repeat,
    roll,
    say,
    score,
    what,
  };
}
