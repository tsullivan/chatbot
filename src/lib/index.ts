export { Log } from './log';
export { Metrics } from './metrics';
export { MultiMap } from './multi_map';
export { roll } from './roll';
export { oneOf } from './one_of';
export { getFamilyTree } from './family_tree';
export { p } from './parajoin';
export { s } from './strip_newlines';
