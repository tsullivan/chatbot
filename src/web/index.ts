import * as apmNode from 'elastic-apm-node';
import * as cookieParser from 'cookie-parser';
import * as express from 'express';
import { BOT_NAME, PORT } from '../constants';
import { Bot } from '../bot';
import { apm as apmConfig } from '../../config';
import { initRoutes } from './routes';
import { initSession } from './session';
import { join } from 'path';

const apm = apmNode.start(apmConfig);

/*
 * Exported for tests
 */
export function getServer(bot: Bot = new Bot()): express.Application {
  const app = express();

  const pubs = join(__dirname, '..', '..', 'public');
  app.use(express.static(pubs)); // home html page, static js

  app.use(cookieParser());
  app.use(apm.middleware.connect());

  app.get('/', (req, res) => {
    res.sendFile(join(pubs, 'index.html'));
  });

  initSession(bot, app);
  initRoutes(bot, app);

  return app;
}

export function runServer(bot: Bot): void {
  const app = getServer(bot);
  const log = bot.getLogger(['web', 'server']);

  app.listen(PORT, () => {
    log.info(['listen'], `${BOT_NAME} listening on [:${PORT}]`);
  });
}
