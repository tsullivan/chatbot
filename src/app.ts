import { argv } from 'yargs';
import { Bot } from './lib/bot';
import { runServer as runWeb } from './server';
import { runBot as runSlack} from './slackbot';

const b = new Bot();
const log = b.getLogger();

log.debug(['start'], 'Bot initialized');

if (argv.web !== 'false') {
  log.info(['web', 'start'], 'Starting web...');
  runWeb(b);
}

if (argv.slack) {
  log.info(['slack', 'start'], 'Starting slack...');
  runSlack(b);
}
