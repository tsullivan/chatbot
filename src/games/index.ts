import { Game as AdventureGame } from './games/adventure';
import { Game as BatmanGame } from './games/batman';
import { Game as BubbleGunGame } from './games/bubble_gun';
import { ChatGame } from './lib';
import { Game as EscapeJailGame } from './games/escape_jail';
import { Game as GuessNumberGame } from './games/guess_number';
import { Game as HikingGame } from './games/hiking';
import { Session } from '../bot';

interface GameSet {
  [gameName: string]: new (session: Session) => ChatGame;
}
export async function getGames(): Promise<GameSet> {
  return {
    adventure: AdventureGame,
    batman: BatmanGame,
    bubble_gun: BubbleGunGame, // eslint-disable-line @typescript-eslint/camelcase
    escape_jail: EscapeJailGame, // eslint-disable-line @typescript-eslint/camelcase
    guess_number: GuessNumberGame, // eslint-disable-line @typescript-eslint/camelcase
    hiking: HikingGame,
  };
}

export { ChatGame } from './lib';
