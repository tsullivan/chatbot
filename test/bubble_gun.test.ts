import * as request from 'supertest';
import { getServer } from '../src/web';
import { utilFactory } from './utils';

const app = getServer();
const agent = request.agent(app);
const { handshake, send, checkResponses } = utilFactory(agent);

describe('bubble_gun', () => {
  beforeEach(() => handshake());

  let resps;
  beforeEach(() => {
    resps = [];
  });

  test('walk around', async () => {
    resps[resps.length] = await send('play bubble_gun');
    resps[resps.length] = await send('look');
    resps[resps.length] = await send('bridge');
    resps[resps.length] = await send('look');
    resps[resps.length] = await send('playground');
    resps[resps.length] = await send('look');
    resps[resps.length] = await send('tree_fort');
    resps[resps.length] = await send('look');
    resps[resps.length] = await send('playground');
    resps[resps.length] = await send('look');
    resps[resps.length] = await send('electronics');
    resps[resps.length] = await send('take_lr41_batteries');
    resps[resps.length] = await send('take_aa_batteries');
    resps[resps.length] = await send('take_led');
    resps[resps.length] = await send('look');
    resps[resps.length] = await send('playground');
    resps[resps.length] = await send('items');
    resps[resps.length] = await send('combine_bubble_gun_and_batteries');
    resps[resps.length] = await send('combine_led_and_batteries');
    resps[resps.length] = await send('items');
    resps[resps.length] = await send('look');
    resps[resps.length] = await send('soaps');
    resps[resps.length] = await send('steal_the_soap');
    resps[resps.length] = await send('combine_bubble_gun_and_soap');
    resps[resps.length] = await send('items');
    resps[resps.length] = await send('playground');
    resps[resps.length] = await send('look');
    resps[resps.length] = await send('magnets');
    resps[resps.length] = await send('explore');
    resps[resps.length] = await send('secret_door');
    resps[resps.length] = await send('examine_control_panel');
    resps[resps.length] = await send('status_cycle');
    resps[resps.length] = await send('status_cycle');
    resps[resps.length] = await send('status_cycle');
    resps[resps.length] = await send('status_cycle');
    resps[resps.length] = await send('status_cycle');
    resps[resps.length] = await send('playground');
    resps[resps.length] = await send('look');
    resps[resps.length] = await send('quit');
    checkResponses(resps);
  });

  test('fall down in soap store', async () => {
    resps[resps.length] = await send('play bubble_gun');
    resps[resps.length] = await send('soaps');
    resps[resps.length] = await send('spill');
    resps[resps.length] = await send('slip');
    resps[resps.length] = await send('spill');
    resps[resps.length] = await send('slip');
    resps[resps.length] = await send('spill');
    resps[resps.length] = await send('slip');
    resps[resps.length] = await send('spill');
    resps[resps.length] = await send('slip');
    resps[resps.length] = await send('quit');
    checkResponses(resps);
  });
});
