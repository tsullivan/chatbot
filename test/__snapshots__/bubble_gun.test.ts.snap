// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bubble_gun fall down in soap store 1`] = `
"# Welcome to Bubble Gun World

You are at: **The Playground**

This playground is right in the center of Bubble Gun World. It has things that are fun for kids, and there's a ladder to a tree fort!!!

Off the corner of the road, there's a bridge leading out of Bubble Gun World.

Nearby is also an Electronics store, a Soaps store, and a Magnets store. I hope you like stores! There are a lot of stores in this game. But guess what: everything in them is FREE!!

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store
- LOOK - Look at the area you are currently in.
- ITEMS - Things you can do with stuff you are holding or stuff you see.
- HELP - See all the commands you can type
- QUIT - Quit the game"
`;

exports[`bubble_gun fall down in soap store 2`] = `
"You are at: **The Soap Store**

Oh my goodness, this store has so much soap! It is by far the soapiest place you have ever been in!

Another thing you notice right away is that there is an extremely strong fragrance of clean in this place.

You see:

- Bottle of liquid soap - This small bottle of liquid soap has more than enough liquid for a bubble gun to shoot out plenty of bubbles.

---

You can:

- STEAL_THE_SOAP - Help yourself to the bottle of liquid soap.
- PLAYGROUND - Go back to the playground
- ACCIDENTALLY_SPILL_SOME_SOAP - You can accidentally spill some soap. No one will blame you because it will be an accident."
`;

exports[`bubble_gun fall down in soap store 3`] = `
"OOPSIE. You accidentally spilled some soap on the floor! Be careful now, the floor might be slippery.

---

You can:

- STEAL_THE_SOAP - Help yourself to the bottle of liquid soap.
- PLAYGROUND - Go back to the playground
- DONT_BE_CAREFUL - You can accidentally not be careful. No one will blame you because it will be another accident."
`;

exports[`bubble_gun fall down in soap store 4`] = `
"OOPSIE. You accidentally weren't careful and slipped on the soap that you spilled on the floor!

BAM! There goes your bottom!

---

You can:

- STEAL_THE_SOAP - Help yourself to the bottle of liquid soap.
- PLAYGROUND - Go back to the playground
- ACCIDENTALLY_SPILL_SOME_MORE_SOAP - You can accidentally spill some soap. No one will blame you because it will be an accident."
`;

exports[`bubble_gun fall down in soap store 5`] = `
"OOPSIE. You accidentally spilled some soap on the floor! Be careful now, the floor might be slippery.

---

You can:

- STEAL_THE_SOAP - Help yourself to the bottle of liquid soap.
- PLAYGROUND - Go back to the playground
- DONT_BE_CAREFUL - You can accidentally not be careful. No one will blame you because it will be another accident."
`;

exports[`bubble_gun fall down in soap store 6`] = `
"OOPSIE. You accidentally weren't careful and slipped on the soap that you spilled on the floor!

BONK! There goes your head!

---

You can:

- STEAL_THE_SOAP - Help yourself to the bottle of liquid soap.
- PLAYGROUND - Go back to the playground
- ACCIDENTALLY_SPILL_SOME_MORE_SOAP - You can accidentally spill some soap. No one will blame you because it will be an accident."
`;

exports[`bubble_gun fall down in soap store 7`] = `
"OOPSIE. You accidentally spilled some soap on the floor! Be careful now, the floor might be slippery.

---

You can:

- STEAL_THE_SOAP - Help yourself to the bottle of liquid soap.
- PLAYGROUND - Go back to the playground
- DONT_BE_CAREFUL - You can accidentally not be careful. No one will blame you because it will be another accident."
`;

exports[`bubble_gun fall down in soap store 8`] = `
"OOPSIE. You accidentally weren't careful and slipped on the soap that you spilled on the floor!

BAM! There goes your bottom!

---

You can:

- STEAL_THE_SOAP - Help yourself to the bottle of liquid soap.
- PLAYGROUND - Go back to the playground
- ACCIDENTALLY_SPILL_SOME_MORE_SOAP - You can accidentally spill some soap. No one will blame you because it will be an accident."
`;

exports[`bubble_gun fall down in soap store 9`] = `
"OOPSIE. You accidentally spilled some soap on the floor! Be careful now, the floor might be slippery.

---

You can:

- STEAL_THE_SOAP - Help yourself to the bottle of liquid soap.
- PLAYGROUND - Go back to the playground
- DONT_BE_CAREFUL - You can accidentally not be careful. No one will blame you because it will be another accident."
`;

exports[`bubble_gun fall down in soap store 10`] = `
"OOPSIE. You accidentally weren't careful and slipped on the soap that you spilled on the floor!

BONK! There goes your head!

---

You can:

- STEAL_THE_SOAP - Help yourself to the bottle of liquid soap.
- PLAYGROUND - Go back to the playground
- ACCIDENTALLY_SPILL_SOME_MORE_SOAP - You can accidentally spill some soap. No one will blame you because it will be an accident."
`;

exports[`bubble_gun fall down in soap store 11`] = `
"I guess you don't want to hike any more. See ya!

Looks like you're a winner! Turns: 9 Score: 50
That was fun, Tim!"
`;

exports[`bubble_gun walk around 1`] = `
"# Welcome to Bubble Gun World

You are at: **The Playground**

This playground is right in the center of Bubble Gun World. It has things that are fun for kids, and there's a ladder to a tree fort!!!

Off the corner of the road, there's a bridge leading out of Bubble Gun World.

Nearby is also an Electronics store, a Soaps store, and a Magnets store. I hope you like stores! There are a lot of stores in this game. But guess what: everything in them is FREE!!

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store
- LOOK - Look at the area you are currently in.
- ITEMS - Things you can do with stuff you are holding or stuff you see.
- HELP - See all the commands you can type
- QUIT - Quit the game"
`;

exports[`bubble_gun walk around 2`] = `
"You are at: **The Playground**

This playground is right in the center of Bubble Gun World. It has things that are fun for kids, and there's a ladder to a tree fort!!!

Off the corner of the road, there's a bridge leading out of Bubble Gun World.

Nearby is also an Electronics store, a Soaps store, and a Magnets store. I hope you like stores! There are a lot of stores in this game. But guess what: everything in them is FREE!!

You see:

- Ladder - This is one sturdy ladder. The bottom part of the ladder is on the ground, and the top part of the ladder leads to the tree fort.

---

You can:

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store"
`;

exports[`bubble_gun walk around 3`] = `
"You are at: **The Bridge Out Of Town**

This long, long bridge carries the road over a very tall gorge that surrounds Bubble Gun World. It is the only way out of town.

There's no way out of town right now, Bubble Gun World Police have blocked the road to prevent anyone from using the road. A power outage in the streets has caused all the street lights to turn off in this area. The Police are saying it's too dark and unsafe to let anyone try to cross.

The police have put up signs all along the bridge that say BRIDGE IS OUT!But it is too dark to even see them.

---

You can:

- PLAYGROUND - Go back to the playground
- CROSS_BRIDGE - Walk over the long, long bridge out of Bubble Gun World"
`;

exports[`bubble_gun walk around 4`] = `
"You are at: **The Bridge Out Of Town**

This long, long bridge carries the road over a very tall gorge that surrounds Bubble Gun World. It is the only way out of town.

There's no way out of town right now, Bubble Gun World Police have blocked the road to prevent anyone from using the road. A power outage in the streets has caused all the street lights to turn off in this area. The Police are saying it's too dark and unsafe to let anyone try to cross.

The police have put up signs all along the bridge that say BRIDGE IS OUT!But it is too dark to even see them.

---

You can:

- PLAYGROUND - Go back to the playground
- CROSS_BRIDGE - Walk over the long, long bridge out of Bubble Gun World"
`;

exports[`bubble_gun walk around 5`] = `
"You are at: **The Playground**

This playground is right in the center of Bubble Gun World. It has things that are fun for kids, and there's a ladder to a tree fort!!!

Off the corner of the road, there's a bridge leading out of Bubble Gun World.

Nearby is also an Electronics store, a Soaps store, and a Magnets store. I hope you like stores! There are a lot of stores in this game. But guess what: everything in them is FREE!!

You see:

- Ladder - This is one sturdy ladder. The bottom part of the ladder is on the ground, and the top part of the ladder leads to the tree fort.

---

You can:

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store"
`;

exports[`bubble_gun walk around 6`] = `
"You are at: **The Playground**

This playground is right in the center of Bubble Gun World. It has things that are fun for kids, and there's a ladder to a tree fort!!!

Off the corner of the road, there's a bridge leading out of Bubble Gun World.

Nearby is also an Electronics store, a Soaps store, and a Magnets store. I hope you like stores! There are a lot of stores in this game. But guess what: everything in them is FREE!!

You see:

- Ladder - This is one sturdy ladder. The bottom part of the ladder is on the ground, and the top part of the ladder leads to the tree fort.

---

You can:

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store"
`;

exports[`bubble_gun walk around 7`] = `
"You are at: **The Tree Fort**

This tree fort is so tall! It has a big window with a clear view of the entire playground.

---

You can:

- PLAYGROUND - Go back to the playground"
`;

exports[`bubble_gun walk around 8`] = `
"You are at: **The Tree Fort**

This tree fort is so tall! It has a big window with a clear view of the entire playground.

---

You can:

- PLAYGROUND - Go back to the playground"
`;

exports[`bubble_gun walk around 9`] = `
"You are at: **The Playground**

This playground is right in the center of Bubble Gun World. It has things that are fun for kids, and there's a ladder to a tree fort!!!

Off the corner of the road, there's a bridge leading out of Bubble Gun World.

Nearby is also an Electronics store, a Soaps store, and a Magnets store. I hope you like stores! There are a lot of stores in this game. But guess what: everything in them is FREE!!

You see:

- Ladder - This is one sturdy ladder. The bottom part of the ladder is on the ground, and the top part of the ladder leads to the tree fort.

---

You can:

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store"
`;

exports[`bubble_gun walk around 10`] = `
"You are at: **The Playground**

This playground is right in the center of Bubble Gun World. It has things that are fun for kids, and there's a ladder to a tree fort!!!

Off the corner of the road, there's a bridge leading out of Bubble Gun World.

Nearby is also an Electronics store, a Soaps store, and a Magnets store. I hope you like stores! There are a lot of stores in this game. But guess what: everything in them is FREE!!

You see:

- Ladder - This is one sturdy ladder. The bottom part of the ladder is on the ground, and the top part of the ladder leads to the tree fort.

---

You can:

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store"
`;

exports[`bubble_gun walk around 11`] = `
"You are at: **The Electronics Store**

Here at the Electronics store, their motto is: \\"Electronics - they're what we sell!\\"

You see:

- Small LR41 Batteries. - The two LR41 batteries are the standard 1.5V button size batteries. They'd fit nicely into the battery holder of a very small piece of electronics.
- Medium-size AA Batteries. - The two AA batteries are the standard 1.5V medium-size batteries. They'd fit nicely into the battery holder of a medium-size piece of electronics.
- Light-emitting Diode - A semiconductor diode which glows when a voltage is applied. It's an LED connected to a circuit board for making it flashy, and battery holder for LR41s.

---

You can:

- TAKE_LR41_BATTERIES - Obtain possession of the LR41 batteries.
- TAKE_AA_BATTERIES - Obtain possession of the AA batteries.
- TAKE_LED - Take the LED.
- PLAYGROUND - Go back to the playground"
`;

exports[`bubble_gun walk around 12`] = `
"The LR41 batteries are now yours!

---

You can:

- TAKE_AA_BATTERIES - Obtain possession of the AA batteries.
- TAKE_LED - Take the LED.
- PLAYGROUND - Go back to the playground"
`;

exports[`bubble_gun walk around 13`] = `
"The AA batteries are now yours!

---

You can:

- TAKE_LED - Take the LED.
- PLAYGROUND - Go back to the playground"
`;

exports[`bubble_gun walk around 14`] = `
"The LED is now yours, and yours alone.

---

You can:

- PLAYGROUND - Go back to the playground"
`;

exports[`bubble_gun walk around 15`] = `
"You are at: **The Electronics Store**

Here at the Electronics store, their motto is: \\"Electronics - they're what we sell!\\"

---

You can:

- PLAYGROUND - Go back to the playground"
`;

exports[`bubble_gun walk around 16`] = `
"You are at: **The Playground**

This playground is right in the center of Bubble Gun World. It has things that are fun for kids, and there's a ladder to a tree fort!!!

Off the corner of the road, there's a bridge leading out of Bubble Gun World.

Nearby is also an Electronics store, a Soaps store, and a Magnets store. I hope you like stores! There are a lot of stores in this game. But guess what: everything in them is FREE!!

You see:

- Ladder - This is one sturdy ladder. The bottom part of the ladder is on the ground, and the top part of the ladder leads to the tree fort.

---

You can:

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store"
`;

exports[`bubble_gun walk around 17`] = `
"In The Playground, You see:
- Ladder - This is one sturdy ladder. The bottom part of the ladder is on the ground, and the top part of the ladder leads to the tree fort.

You are holding:
- Bubble Gun - It's a bubble gun for gunning out bubbles. If you have soap and two AA batteries, it can fulfill the purpose of its existence.
  - COMBINE_BUBBLE_GUN_AND_BATTERIES - Put the AA batteries in the bubble gun.
- Small LR41 Batteries. - The two LR41 batteries are the standard 1.5V button size batteries. They'd fit nicely into the battery holder of a very small piece of electronics.
- Medium-size AA Batteries. - The two AA batteries are the standard 1.5V medium-size batteries. They'd fit nicely into the battery holder of a medium-size piece of electronics.
- Light-emitting Diode - A semiconductor diode which glows when a voltage is applied. It's an LED connected to a circuit board for making it flashy, and battery holder for LR41s.
  - COMBINE_LED_AND_BATTERIES - Put the two LR41 batteries in the battery holder of the LED device."
`;

exports[`bubble_gun walk around 18`] = `
"You slide off the cover to the bubble gun, pop in the AA batteries, and put the cover back on. It's nice and snug with the water seal that will prevent corrosion in the batteries or the metal surfaces inside the bubble gun's battery compartment.

---

You can:

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store"
`;

exports[`bubble_gun walk around 19`] = `
"The batteries go right on in to the LED's battery holder. Nice and simple.

---

You can:

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store"
`;

exports[`bubble_gun walk around 20`] = `
"In The Playground, You see:
- Ladder - This is one sturdy ladder. The bottom part of the ladder is on the ground, and the top part of the ladder leads to the tree fort.

You are holding:
- Bubble Gun - Bubble gun with batteries, but no soap.
- Light-emitting Diode - LED throwie that has batteries, but you can't do anything else with it."
`;

exports[`bubble_gun walk around 21`] = `
"You are at: **The Playground**

This playground is right in the center of Bubble Gun World. It has things that are fun for kids, and there's a ladder to a tree fort!!!

Off the corner of the road, there's a bridge leading out of Bubble Gun World.

Nearby is also an Electronics store, a Soaps store, and a Magnets store. I hope you like stores! There are a lot of stores in this game. But guess what: everything in them is FREE!!

You see:

- Ladder - This is one sturdy ladder. The bottom part of the ladder is on the ground, and the top part of the ladder leads to the tree fort.

---

You can:

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store"
`;

exports[`bubble_gun walk around 22`] = `
"You are at: **The Soap Store**

Oh my goodness, this store has so much soap! It is by far the soapiest place you have ever been in!

Another thing you notice right away is that there is an extremely strong fragrance of clean in this place.

You see:

- Bottle of liquid soap - This small bottle of liquid soap has more than enough liquid for a bubble gun to shoot out plenty of bubbles.

---

You can:

- STEAL_THE_SOAP - Help yourself to the bottle of liquid soap.
- PLAYGROUND - Go back to the playground
- ACCIDENTALLY_SPILL_SOME_SOAP - You can accidentally spill some soap. No one will blame you because it will be an accident."
`;

exports[`bubble_gun walk around 23`] = `
"Did you forget that I said everything in this town is FREE!? Oh well. You wanted the soap, you got it. It's your problem now.

---

You can:

- PLAYGROUND - Go back to the playground
- ACCIDENTALLY_SPILL_SOME_SOAP - You can accidentally spill some soap. No one will blame you because it will be an accident."
`;

exports[`bubble_gun walk around 24`] = `
"You slide off the cover to the bubble gun, pop in the AA batteries, and put the cover back on. It's nice and snug with the water seal that will prevent corrosion in the batteries or the metal surfaces inside the bubble gun's battery compartment.

The bubble gun is completely functional now!

---

You can:

- PLAYGROUND - Go back to the playground
- ACCIDENTALLY_SPILL_SOME_SOAP - You can accidentally spill some soap. No one will blame you because it will be an accident."
`;

exports[`bubble_gun walk around 25`] = `
"You are holding:
- Bubble Gun - Completely functional and working bubble gun
  - BUBBLE_BLASTER - Blast bubbles out the bubble gun
- Light-emitting Diode - LED throwie that has batteries, but you can't do anything else with it."
`;

exports[`bubble_gun walk around 26`] = `
"You are at: **The Playground**

This playground is right in the center of Bubble Gun World. It has things that are fun for kids, and there's a ladder to a tree fort!!!

Off the corner of the road, there's a bridge leading out of Bubble Gun World.

Nearby is also an Electronics store, a Soaps store, and a Magnets store. I hope you like stores! There are a lot of stores in this game. But guess what: everything in them is FREE!!

You see:

- Ladder - This is one sturdy ladder. The bottom part of the ladder is on the ground, and the top part of the ladder leads to the tree fort.

---

You can:

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store"
`;

exports[`bubble_gun walk around 27`] = `
"You are at: **The Playground**

This playground is right in the center of Bubble Gun World. It has things that are fun for kids, and there's a ladder to a tree fort!!!

Off the corner of the road, there's a bridge leading out of Bubble Gun World.

Nearby is also an Electronics store, a Soaps store, and a Magnets store. I hope you like stores! There are a lot of stores in this game. But guess what: everything in them is FREE!!

You see:

- Ladder - This is one sturdy ladder. The bottom part of the ladder is on the ground, and the top part of the ladder leads to the tree fort.

---

You can:

- BRIDGE - Go to the bridge
- TREE_FORT - Climb the ladder up to the tree fort
- ELECTRONICS - Go to the Electronics store
- SOAPS - Go to the Soaps store
- MAGNETS - Go to the Magnet store"
`;

exports[`bubble_gun walk around 28`] = `
"You are at: **The Magnet Store**

There are lots and lots of metal shelves in here. The entire surfaces of all the shelves have magnets stuck to them. The aisles and aisles of shelves create a mysterious atmosphere, where anything could be hiding. Perhaps if you explore a bit, you could find there is more to this store.

You see:

- Small magnet - This looks like a pretty handy little small magent! It has a sticker on for sticking onto a small device.

---

You can:

- TAKE_MAGNET - Take the magnet.
- PLAYGROUND - Go back to the playground
- EXPLORE - See what could be hiding in this strange shop."
`;

exports[`bubble_gun walk around 29`] = `
"The magnet store has the feel of a place where anything could be hiding. It's time to get to the bottom of this mystery.

You wander around some aisles, go around some corners, look around. You're not exactly sure what you're looking for. You are keeping your eyes open for anything that seems out of place in this strange store.

Just when you think maybe this is a normal magnet store after all, you discover a small door on a wall. It is almost completely hidden behind a shelf leaning against the wall.

---

You can:

- TAKE_MAGNET - Take the magnet.
- PLAYGROUND - Go back to the playground
- SECRET_DOOR - Sneak into the small secret door on the wall behind the leaning shelf."
`;

exports[`bubble_gun walk around 30`] = `
"The door leads directly to a case of stairs going down. There are many stairs to walk down, and it takes time to get all the way to the bottom. Finally, you make it to the bottom where there is a room with a wide, open floor, and a very tall ceiling.

You are at: **The Secret Basement of the Magnet Store**

You're at the bottom of a long set of stairs, in a room with a wide, open floor, and a very tall ceiling.

A metal box on the wall stands out as it is the only thing that makes out a shape in the room. The section of wall with the box is protected by a guard rail, which you can walk behind to access the control panel. There are many magnets stuck to the box and the guard rail, just like the shelves upstairs.

It is a very well-lit and clean place - it almost feels sterile.

---

You can:

- UP - Go back to the magnet store
- EXAMINE_CONTROL_PANEL - Take a look at what this control panel has to offer.
- STATUS_CYCLE - Press the STATUS_CYCLE button"
`;

exports[`bubble_gun walk around 31`] = `
"The panel has a small screen and a button of about the same size. The screen reads one message: **STATUS: LOOMING**. The button is labeled **STATUS_CYCLE**. There is another slightly larger screen on the control panel: a map showing all of Bubble Gun World. It shows targets on the Playground and the Bridge.

---

You can:

- UP - Go back to the magnet store
- STATUS_CYCLE - Press the STATUS_CYCLE button"
`;

exports[`bubble_gun walk around 32`] = `
"The bright florescent lights suddently shut off. Red alerting lights turn on and paint the large room with red light and long shadows.

The floor starts to shake, and it becomes necessary for you to grab onto the guard rail to remain standing. The floor starts to separate in the middle - opening like cargo doors - to reveal machinery beneath the floor.

In the glow of the red light, you can still read the message on the control panel status screen. It has changed to: **STATUS: EXPOSED**

---

You can:

- UP - Go back to the magnet store
- STATUS_CYCLE - Press the STATUS_CYCLE button"
`;

exports[`bubble_gun walk around 33`] = `
"Gears in the machinery start to slowly turn, and parts of the machine start to unfold. First, metal arms and cranks reach out of the separated floor, into the air, higher and higher. It crashes the ceiling, and creates a new window into the Magnet Store above. Magnets are wapping off the shelves and wapping onto the machine arm. But the machine arm still goes up and up and up. It crashes out of the roof of the building, and then finally becomes still.

All the crashing has created a lot of debris, rubble and dust in the basement around you. But the control panel is still working. Now the status screen reads: **STATUS: UNDEFINED**

It seems this part of the machine functionality has never been tested before. That would explain why the ceilings and roof wasn't destroyed when you got into town.

The Playground and the Bridge on the map of the control panel have their targets blinking.

---

You can:

- UP - Go back to the magnet store
- STATUS_CYCLE - Press the STATUS_CYCLE button"
`;

exports[`bubble_gun walk around 34`] = `
"Suddenly, a loud noise blasts your ears: THUDUDUDUDUDUDUDUDUDUDUDUDUDUDUDUDUD. The noise is coming from high above you, outside the building. After about a minute, the noise stops.

The Playground's icon on the map changed from a target to an X.

---

You can:

- UP - Go back to the magnet store
- STATUS_CYCLE - Press the STATUS_CYCLE button"
`;

exports[`bubble_gun walk around 35`] = `
"Again a loud noise blasts your ears: THUDUDUDUDUDUDUDUDUDUDUDUDUDUDUDUDUD. It's just as loud and long as last time. This time, when the THUDs stop, you hear some clicks, and then that's it.

The Bridge's icon on the map changed from a target to an X.

---

You can:

- UP - Go back to the magnet store
- STATUS_CYCLE - Press the STATUS_CYCLE button"
`;

exports[`bubble_gun walk around 36`] = `
"When you push the button now, the machine shakes a little bit and makes some clunking sounds, but nothing happens. It seems to be out of ammunition.

---

You can:

- UP - Go back to the magnet store
- STATUS_CYCLE - Press the STATUS_CYCLE button"
`;

exports[`bubble_gun walk around 37`] = `
"ERROR! LOSE 2 POINTS. Type HELP to show all the commands

- UP - Go back to the magnet store
- STATUS_CYCLE - Press the STATUS_CYCLE button
- LOOK - Look at the area you are currently in.
- ITEMS - Things you can do with stuff you are holding or stuff you see.
- HELP - See all the commands you can type
- QUIT - Quit the game"
`;

exports[`bubble_gun walk around 38`] = `
"You are at: **The Secret Basement of the Magnet Store**

You're at the bottom of a long set of stairs, in a room with a wide, open floor, and a very tall ceiling.

A metal box on the wall stands out as it is the only thing that makes out a shape in the room. The section of wall with the box is protected by a guard rail, which you can walk behind to access the control panel. There are many magnets stuck to the box and the guard rail, just like the shelves upstairs.

It is a very well-lit and clean place - it almost feels sterile.

---

You can:

- UP - Go back to the magnet store
- STATUS_CYCLE - Press the STATUS_CYCLE button"
`;

exports[`bubble_gun walk around 39`] = `
"I guess you don't want to hike any more. See ya!

Looks like you're a winner! Turns: 37 Score: 48
That was fun, Tim!"
`;
